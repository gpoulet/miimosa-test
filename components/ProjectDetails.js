import { string, number, shape } from "prop-types"

function ProjectDetails({ id, title, collected, target, status }) {
  return (
    <div>
      <p>
      id: {id}<br/>
      title: {title}<br/>
      collected: {collected}<br/>
      target: {target}<br/>
      status: {status}<br/>
      </p>
    </div>
  )
}

ProjectDetails.propTypes = {
  id: string.isRequired,
  title: string.isRequired,
  collected: number.isRequired,
  target: number.isRequired,
  status: string.isRequired,
  description: string.isRequired,
  imageUrl: string.isRequired,
  user: shape({
    first_name: string,
    last_name: string
  }),
};

export default ProjectDetails;

//
// "collected": 15000,
//   "description": "",
//   "id": "c6d6f6f9-07d8-4540-87a7-3b8ff8869f55",
//   "image_url": "",
//   "status": "success",
//   "target": 15000,
//   "title": "Bar smoothie Bio",
//   "user": {
//     "first_name": "Lancelot",
//     "last_name": "Little Finger"
// }