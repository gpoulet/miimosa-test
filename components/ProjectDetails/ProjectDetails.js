import { string, number, shape } from "prop-types"
import Description from '../Description'
import Status from '../Status'
import Title from '../Title'
import Money from '../Money'
import Card from "@material-ui/core/Card"
import CardHeader from '@material-ui/core/CardHeader'
import Avatar from '@material-ui/core/Avatar'
import CardContent from '@material-ui/core/CardContent'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'

function ProjectDetails({ title, collected, target, status, user }) {
  const showParticipateButton = status === "in_progress"
  return (
    <Card className="projectCardDetails">
      <CardHeader
        avatar={<Avatar aria-label="recipe">{user.first_name[0]}</Avatar>}
        title={user.first_name}
        subheader={user.last_name}
        action={<Status status={status}/>}
      />
      <CardContent>
        <Title title={title}/>
        <Money collected={collected} target={target} status={status}/>
        <Description/>
      </CardContent>
      {
       showParticipateButton &&
      <CardActions>
        <Button size="small" color="primary">
          <a>Je veux participer</a>
        </Button>
      </CardActions>
      }
    </Card>
  )
}

ProjectDetails.propTypes = {
  id: string.isRequired,
  title: string.isRequired,
  collected: number.isRequired,
  target: number.isRequired,
  status: string.isRequired,
  description: string.isRequired,
  imageUrl: string,
  user: shape({
    first_name: string.isRequired,
    last_name: string.isRequired
  })
};

export default ProjectDetails;

//
// "collected": 15000,
//   "description": "",
//   "id": "c6d6f6f9-07d8-4540-87a7-3b8ff8869f55",
//   "image_url": "",
//   "status": "success",
//   "target": 15000,
//   "title": "Bar smoothie Bio",
//   "user": {
//     "first_name": "Lancelot",
//     "last_name": "Little Finger"
// }