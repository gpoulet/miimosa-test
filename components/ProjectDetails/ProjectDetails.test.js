import React from 'react';
import renderer from 'react-test-renderer';
import ProjectDetails from './ProjectDetails';

it('ProjectDetails in progress renders correctly', () => {
  const tree = renderer
    .create(
      <ProjectDetails
      id="test"
      status="in_progress"
      collected={15000}
      target={25000}
      description="coucou"
      title="Boulangerie bio"
      imageUrl="/myUrl.png"
      user={{ first_name: "Harry", last_name: "Potter"}}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ProjectDetails failed renders correctly', () => {
  const tree = renderer
    .create(
      <ProjectDetails
      id="test"
      status="failed"
      collected={15}
      target={8000}
      description="venez"
      title="Brasserie du midi"
      imageUrl="/myBeer.png"
      user={{ first_name: "Lucius", last_name: "Malefoy"}}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ProjectDetails success renders correctly', () => {
  const tree = renderer
    .create(
      <ProjectDetails
      id="test"
      status="success"
      collected={10000}
      target={8000}
      description="venez"
      title="Cheese'up Fromagerie responsable"
      imageUrl="/chees.png"
      user={{ first_name: "Hermione", last_name: "Granger"}}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});