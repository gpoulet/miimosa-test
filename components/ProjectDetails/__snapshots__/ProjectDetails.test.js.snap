// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails failed renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCardDetails MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        L
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Lucius
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Malefoy
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <div
        className="MuiChip-root statusChip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          Echec
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiCardMedia-root"
    style={
      Object {
        "backgroundImage": "url(\\"/images/test.png\\")",
        "height": 0,
        "paddingTop": "40%",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Brasserie du midi
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      15€ collectés sur 8000€ (0%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-99.8125%)",
          }
        }
      />
    </div>
    <p
      className="MuiTypography-root description MuiTypography-body1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</div>
`;

exports[`ProjectDetails in progress renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCardDetails MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        H
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Harry
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Potter
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <div
        className="MuiChip-root statusChip MuiChip-colorSecondary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          En cours
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiCardMedia-root"
    style={
      Object {
        "backgroundImage": "url(\\"/images/test.png\\")",
        "height": 0,
        "paddingTop": "40%",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Boulangerie bio
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      15000€ collectés sur 25000€ (60%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={60}
      className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-40%)",
          }
        }
      />
    </div>
    <p
      className="MuiTypography-root description MuiTypography-body1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a>
          Je veux participer
        </a>
      </span>
    </button>
  </div>
</div>
`;

exports[`ProjectDetails success renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root projectCardDetails MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        H
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Hermione
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Granger
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <div
        className="MuiChip-root statusChip MuiChip-colorPrimary"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label"
        >
          Succès
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiCardMedia-root"
    style={
      Object {
        "backgroundImage": "url(\\"/images/test.png\\")",
        "height": 0,
        "paddingTop": "40%",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Cheese'up Fromagerie responsable
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      10000€ collectés sur 8000€ (125%)
    </p>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(0%)",
          }
        }
      />
    </div>
    <p
      className="MuiTypography-root description MuiTypography-body1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</div>
`;
